### PR TITLE
CIVIL143: Stylin' Email Flows

### PR DESCRIPTION
**Pull request by jorgelo**
*** Please note there are some todos and styling fixes that still need to be completed. I am creating this PR a bit prematurely since its a large change and it's better for us to have this merged in right now. ***

* Styles the flows to match the design
* Fixes a bunch of bugs

### CODE REVIEW

File: packages/components/src/Account/Auth/AuthStyledComponents.tsx [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  border-top: 1px solid #d8d8d8;
+  margin-top: 40px;
+  padding-top: 20px;
+`;
+
+export const BenefitsLink = styled(AuthPageFooterLink)`
+  padding: 0;
+`;
+
+export const AuthFooterTerms = () => (
```
Comment (sruddy) [inline packages/components/src/Account/Auth/AuthStyledComponents.tsx]: We've been creating a separate file for text components even if it's just one or two `AuthTextComponents.tsx`

File: packages/components/src/Account/Auth/AuthStyledComponents.tsx [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+export const AuthWrapper: React.SFC = ({ children }) => (
+  <AuthOuterWrapper>
+    <AuthInnerWrapper>{children}</AuthInnerWrapper>
+    <AuthFooterTerms />
+  </AuthOuterWrapper>
+);
+
+export const AuthInnerWrapper = styled.div`
+  margin: 0 115px;
+  margin-top: 71px;
```
Comment (sruddy) [inline packages/components/src/Account/Auth/AuthStyledComponents.tsx]: These can be combine to save a byte

File: packages/components/src/Account/Auth/EmailAuth.tsx [hunk 3]
```diff
@@ -30,68 +34,129 @@ export interface AccountEmailAuthProps {
...
-  errorMessage: string | null;
+  errorMessage: string | undefined;
+  hasAgreedToTOS: boolean;
+  hasSelectedToAddToNewsletter: boolean;
+  hasBlurred: boolean;
+}
+
+// TODO(jorgelo): Is there a more official way to do this?
+
+function isValidEmail(email: string): boolean {
```
Comment (sruddy) [inline packages/components/src/Account/Auth/EmailAuth.tsx]: You should add `isValidEmail ` as a util if there isn't one already. Also there is some regex in utils you can either add the email or use what's there if already available.

File: packages/components/src/Account/Auth/AuthStyledComponents.tsx [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+  margin: 30px 0;
+`;
+
+export const CenterWrapper: React.SFC = ({ children }) => (
+  <div style={{ display: "flex", justifyContent: "center" }}>
+    <div>{children}</div>
+  </div>
+);
+
+export const CheckEmailWrapper = styled.ul``;
```
Comment (sruddy) [inline packages/components/src/Account/Auth/AuthStyledComponents.tsx]: Is this being used? If there are no styles on this you should just use ul.

File: packages/components/src/Account/Auth/EmailSent.tsx [hunk 5]
```diff
@@ -1,19 +1,35 @@
...
 export class AccountEmailSent extends React.Component<AccountEmailSentProps> {
   public render(): JSX.Element {
-    const emailAddress = this.props.emailAddress || this.props.location!.state.emailAddress;
+    const { emailAddress, onSendAgain } = this.props;
 
-    const { isNewUser } = this.props;
     return (
       <>
-        Check your email to {!isNewUser ? "sign in" : "register"}: <strong>{emailAddress}</strong>
+        <PageSubHeadingCentered>Check your email!</PageSubHeadingCentered>
```
Comment (sruddy) [inline packages/components/src/Account/Auth/EmailSent.tsx]: You can add this text component to the text file also

File: packages/components/src/Account/Auth/VerifyToken.tsx [hunk 6]
```diff
@@ -52,14 +48,28 @@ export class AccountVerifyToken extends React.Component<AccountVerifyTokenProps,
...
     const errorText = error.graphQLErrors.map((e: any) => e.message).join(" ,");
 
     return <>{errorText}</>;
   }
 
+  public renderVerified(): JSX.Element {
+    const { onAuthenticationContinue, isNewUser } = this.props;
+    return (
+      <>
+        <PageSubHeadingCentered>Email Address Confirmed!</PageSubHeadingCentered>
```
Comment (sruddy) [inline packages/components/src/Account/Auth/VerifyToken.tsx]: Same with this text

File: packages/components/src/Heading.tsx [hunk 7]
```diff
@@ -62,3 +61,38 @@ export const ManagerSectionHeading = styled.h4`
 `;
 
 ManagerSectionHeading.defaultProps = headingDefaultProps;
+
+export const PageHeadingCentered = styled.h2`
+  font-family: ${props => props.theme.sansSerifFont};
```
Comment (sruddy) [inline packages/components/src/Heading.tsx]: for fonts you should use the style constant ${fonts.SANS_SERIF}

File: packages/dapp/src/components/Auth/Login.tsx [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+  AuthPageFooterLink,
+  AuthWrapper,
+} from "@joincivil/components";
+import { Link } from "react-router-dom";
+
+export interface AuthLoginProps {
+  onEmailSend(isNewUser: boolean, emailAddress: string): void;
+}
+
+export const AuthLogin: React.SFC<AuthLoginProps> = props => {
```
Comment (sruddy) [inline packages/dapp/src/components/Auth/Login.tsx]: Same with this text

File: packages/dapp/src/components/Auth/Signup.tsx [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+} from "@joincivil/components";
+import { Link } from "react-router-dom";
+
+export interface AuthSignupProps {
+  onEmailSend(isNewUser: boolean, emailAddress: string): void;
+}
+
+export const AuthSignup: React.SFC<AuthSignupProps> = props => {
+  return (
+    <AuthWrapper>
```
Comment (sruddy) [inline packages/dapp/src/components/Auth/Signup.tsx]: ditto with this text

File: packages/components/src/Account/Auth/AuthStyledComponents.tsx (hunk 1 above)
Comment (jorgelo) [reply packages/components/src/Account/Auth/AuthStyledComponents.tsx]: Done, created AuthTextComponents.

File: packages/components/src/Account/Auth/AuthStyledComponents.tsx (hunk 2 above)
Comment (jorgelo) [reply packages/components/src/Account/Auth/AuthStyledComponents.tsx]: Good call.

File: packages/components/src/Account/Auth/EmailAuth.tsx (hunk 3 above)
Comment (jorgelo) [reply packages/components/src/Account/Auth/EmailAuth.tsx]: Done, created it in utils.

File: packages/components/src/Account/Auth/AuthStyledComponents.tsx (hunk 4 above)
Comment (jorgelo) [reply packages/components/src/Account/Auth/AuthStyledComponents.tsx]: Nope, good catch!

File: packages/components/src/Account/Auth/EmailSent.tsx (hunk 5 above)
Comment (jorgelo) [reply packages/components/src/Account/Auth/EmailSent.tsx]: Done.

File: packages/dapp/src/components/Auth/Signup.tsx (hunk 9 above)
Comment (jorgelo) [reply packages/dapp/src/components/Auth/Signup.tsx]: Done.

File: packages/dapp/src/components/Auth/Login.tsx (hunk 8 above)
Comment (jorgelo) [reply packages/dapp/src/components/Auth/Login.tsx]: Done.

File: packages/components/src/Account/Auth/EmailAuth.tsx [hunk 10]
```diff
@@ -101,6 +160,10 @@ export class AccountEmailAuth extends React.Component<AccountEmailAuthProps, Acc
 
     if (authResponse === "ok") {
       onEmailSend(isNewUser, emailAddress);
+      if (hasSelectedToAddToNewsletter) {
```
Comment (dankins) [inline packages/components/src/Account/Auth/EmailAuth.tsx]: is there a Jira ticket for this? Want to make sure it doesn't get lost

File: packages/components/src/input/checkbox.stories.tsx [hunk 11]
```diff
@@ -41,6 +41,13 @@ storiesOf("check box", module)
   .add("slide check box", () => {
     return <TestSlideComponent />;
   })
-  .add("checkbox", () => {
-    return <TestComponent />;
-  });
+  .add("checkboxs", () => (
```
Comment (dankins) [inline packages/components/src/input/checkbox.stories.tsx]: typo

File: packages/components/src/Account/Auth/EmailAuth.tsx (hunk 10 above)
Comment (jorgelo) [reply packages/components/src/Account/Auth/EmailAuth.tsx]: Good call, I created CIVIL-381 and added a note in the code.

File: packages/components/src/input/checkbox.stories.tsx (hunk 11 above)
Comment (jorgelo) [reply packages/components/src/input/checkbox.stories.tsx]: Fixed, and made it a little prettier while I was at it.

**Pull request merged**